superbench/analyzer/result_summary.py

```diff
@@ -0,0 +1,244 @@
+
+        Returns:
+            DataFrame: summary of all rules
+        """
+        summary_df = pd.DataFrame()
+        for category in summary:
+            for i in range(len(summary[category])):
+                summary_df = summary_df.append([summary[category][i]], ignore_index=True)
+        return summary_df
+
+    def _generate_summary(self, round):
+        r"""Generate summay dict of all rules.
+
+        For each rule, aggregate the data by user-defined pattern or ranks (:\\d+), calculate
+        the list of statistics of aggregated metrics, then format the summary in {category, lines}.
+
+        Args:
+            round (int): the number of decimal digits
+
+        Returns:
+            dict: summary dict, the keys are categories, the values are summary lines for the category
+        """
+        summary = {}
+        for rule in self._sb_rules:
+            metrics = list(self._sb_rules[rule]['metrics'].keys())
+            category = self._sb_rules[rule]['categories']
+            data_df_of_rule = self._raw_data_df[metrics]
+            if self._sb_rules[rule]['aggregate']:
+                # if aggregate is True, aggregate in ranks
+                if self._sb_rules[rule]['aggregate'] is True:
+                    data_df_of_rule = data_analysis.aggregate(data_df_of_rule)
+                # if aggregate is not empty and is a pattern in regex, aggregate according to pattern
+                else:
+                    data_df_of_rule = data_analysis.aggregate(data_df_of_rule, self._sb_rules[rule]['aggregate'])
+            statistics = self._sb_rules[rule]['statistics']
+            summary_df_of_rule = pd.DataFrame(columns=sorted(data_df_of_rule.columns))
+            for statistic_name in statistics:
+                # get SummaryOp and calculate statistics
+                # if statistic_name is 'p\d\d?', SummaryOp should be pencentile
+                if str.startswith(statistic_name, 'p'):
+                    rule_op = SummaryOp.get_summary_func(SummaryType('percentile'))
+                    val = int(statistic_name.strip('p'))
+                    summary_df_of_rule.loc[statistic_name] = rule_op(data_df_of_rule, val)
+                else:
+                    rule_op = SummaryOp.get_summary_func(SummaryType(statistic_name))
+                    summary_df_of_rule.loc[statistic_name] = rule_op(data_df_of_rule)
+            # format values to n significant decimal digits
+            if round and isinstance(round, int):
+                summary_df_of_rule = data_analysis.round_significant_decimal_places(
+                    summary_df_of_rule, round, list(summary_df_of_rule.columns)
+                )
+            # format summary_df of a rule to list of lines
+            summary_lines_of_rule = self._format_summary_of_rule(category, summary_df_of_rule)
+            summary[category] = summary_lines_of_rule
+
+        return summary
+
+    def gen_md_lines(self, summary):
```

Comment:
why some function name use `generate`, and other use `gen`?

Reply:
unify to generate